packages/gatsby-plugin-google-analytics/src/index.js

```diff
@@ -53,4 +53,32 @@ OutboundLink.propTypes = {
   onClick: PropTypes.func,
 }
 
-export { OutboundLink }
+/**
+ * This allows the user to create custom events within their Gatsby projects.
+ * @param       {string} category Required - The object that was interacted with (e.g.video)
+ * @param       {string} action   Required - Type of interaction (e.g. 'play')
+ * @param       {string} label    Optional - Useful for categorizing events (e.g. 'Spring Campaign')
+ * @param       {string} value    Optional - Numeric value associated with the event. (e.g. A product ID)
+ *
+ * @link        https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#events
+ */
+
+function CustomTrackingEvent({ category, action, label, value }) {
```

Comment:
trackCustomEvent is a better name as in javascript we don't like to start things with a capital letter unless it is a class.

```suggestion
function trackCustomEvent({ category, action, label, value }) {
```